Nullablity warnings fixes, ConfigureAwait(false) Updates, Doc Comment Updates.

mohummedibrahim opened the pull request:
1) Nullability Warning Fixes:
- Added the `#nullable disable` to API result models to get rid of nullable warnings. 
- For some specific properties either added the `string.Empty` or appended `?` based on their chances to be nullable.

2) `ConfigureAwait(false)` Updates: Added the `ConfigureAwait(false)` to all awaited tasks.

3) Doc Comment Updates:
- Added the Doc comments to all endpoints from Jotform API Documentation. 
- In a few places, for now, I've just added `TODO:` where I thought, later on, those input parameters could be strongly typed, or at least documented with proper details.

mohummedibrahim committed:
- Added the `#nullable disable` to Api result model to get ride of nullable warnings, for some properties either added the `string.Empty` or appended `?`.

mohummedibrahim committed:
Added the `.ConfigureAwait(false)` to all awaited tasks.

mohummedibrahim committed:
Added the Doc comments to all endpoints from Jotform API Documentation. In a few places, for now I've just added `TODO:` where I thought, later on, those input parameter could be strongly typed, or at least documented with proper details.

Hona reviewed (changes requested):
May I ask why there is a bunch of 

```
#nullable disable
```

Is it possible to use correct null annotations, or if you're unsure just mark it as a nullable property for now?
Unless there is a good reason to?

mohummedibrahim commented:
> May I ask why there is a bunch of
> 
> ```
> #nullable disable
> ```
> 
> Is it possible to use correct null annotations, or if you're unsure just mark it as a nullable property for now? Unless there is a good reason to?

Yes, that's what my first attempt was, marking the strings as nullable, but it was making the library user check null for every (nullable) string even though those would always be non-nullable (based on the sample response). Soon I realized the response models you'd already written were exactly matching with Jotform API sample responses in regards to nullability. So I tried not to alter them.

Another option I had `string.Empty`, it seemed to be alerting the behavior, for example, something we believe won't be null but somehow it's, the Null Exception won't be thrown. I've used this at a couple of places where I was sure it wouldn't have any side effects. 

One more option I had is null forgiving e.g. `null!` value, I've seen this is mostly used with EFCore or serializing data objects and has mostly the same effect as `#nullable disable`. So I ended up adding this just as a workaround for now. 

Would you like me to annotate these with `null!`? or please feel free to let me know which option you think is suitable? Or perhaps skip the Nullabilty comment for now?

Hona commented:
If you don't mind, could you change it to `= null!`.
That way we still mark the fields as not null, but implicitly say that the JSON deserialization will set the required fields.
Because we're deserializing from JSON we are kind of walking around normal object initialization.

Hona commented:
Once you do, the PR is perfect to merge. Great work.

Something I noticed - is there a reason these methods are public, and not private:

![image](https://github.com/user-attachments/assets/accade9c-9c77-4c66-ae5b-a4c3440040a0)

mohummedibrahim commented:
> If you don't mind, could you change it to `= null!`. That way we still mark the fields as not null, but implicitly say that the JSON deserialization will set the required fields. Because we're deserializing from JSON we are kind of walking around normal object initialization.

Yes, no worries at all, I'll change it to `= null!`.

mohummedibrahim commented:
> Once you do, the PR is perfect to merge. Great work.
> 
> Something I noticed - is there a reason these methods are public, and not private:
> 
> ![image](https://private-user-images.githubusercontent.com/10430890/352881638-accade9c-9c77-4c66-ae5b-a4c3440040a0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIyNzEwNzQsIm5iZiI6MTcyMjI3MDc3NCwicGF0aCI6Ii8xMDQzMDg5MC8zNTI4ODE2MzgtYWNjYWRlOWMtOWM3Ny00YzY2LWFlNWItYTRjMzQ0MDA0MGEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI5VDE2MzI1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFjY2Q4ZjM5Y2FjYTNhNTlhMDEzYzdhODI3ZWNlYjQ5NjI2YTYzMjNiNzk5OWQyMTEyZTI2NTkzYzhlNTY3MjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8EOe3F_2tQFPOGnm5tFQn4n31KlnRLxvLBTkiilX0FA)

Nope, no reason, they're actually supposed to be private, I'll change their accessibility.

mohummedibrahim committed:
- Removed the `#nullable disable` directives from types and added the null forgiving operators `null!` to the properties (mostly string properties)
-  In some places updated the properties to be nullable (i.e. appended `?').
- Changed the `FormReport.UpdatedAt` property type from object to string?

mohummedibrahim committed:
- Changed the accessiblity of `GetResultAsync` and `GetPagedResultAsync` methods to private.

mohummedibrahim commented:
Updated the PR with changes, please let me know if you find any other issues. Thanks!

Hona commented:
Great stuff! I'll approve and merge
